app/controllers/data_sets_controller.rb

```diff
@@ -305,16 +327,59 @@ def dataFileUpload
       end
     end
 
-    if params[:file]
-      @filename = DataSet.get_next_name(project, params[:file].original_filename.split('.')[0])
-    else
-      @filename = DataSet.get_next_name(project, 'Google docs')
+    if params[:file] && params[:file].original_filename.split('.')[1] == 'zip'
```

Comment:
might have problems with filenames like `dogs.cats.zip` -> will extract `cats` as the filetype, not `zip`.  Try `-1` instead of `1`.